projects/cds/src/merchandising/facade/cds-merchandising-product.service.ts

```diff
@@ -18,31 +33,74 @@ export class CdsMerchandisingProductService {
     protected strategyConnector: MerchandisingStrategyConnector,
     protected baseSiteService: BaseSiteService,
     protected languageService: LanguageService,
-    private routingService: RoutingService
+    private routingService: RoutingService,
```

Comment:
This is starting to be a big facade with a lot of injected services. It makes testing a harder (need to mock a lot dependencies) and you have to test more use case combinations.
I suggest you split it to two smaller services. What's the best logical way to split it? Maybe move what was in this facade to one service, and the second service can have everything related to categories/facets?

After that, just inject both services in this facade and combine observables.

Reply:
I've split the functionality into site and user context services instead